src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -61,11 +62,15 @@ private static string MarshalToString(IntPtr arg, string argName)
                 string typeName = MarshalToString(typeNameNative, nameof(typeNameNative));
                 string methodName = MarshalToString(methodNameNative, nameof(methodNameNative));
 
-                string delegateType;
+                string? delegateType;
                 if (delegateTypeNative == IntPtr.Zero)
                 {
                     delegateType = typeof(ComponentEntryPoint).AssemblyQualifiedName!;
```

Comment:
Getting the type name and immediately converting it back into a Type is waste of cycles.